core/src/main/java/org/modelmapper/internal/ExplicitMappingVisitor.java

```diff
@@ -329,12 +330,41 @@ private Method methodFor(Class<?> type, Type methodType, MethodInsnNode mn) {
     Type[] argumentTypes = methodType.getArgumentTypes();
     Class<?>[] paramTypes = new Class<?>[argumentTypes.length];
     for (int i = 0; i < paramTypes.length; i++) {
-      Class<?> paramType = Primitives.primitiveFor(argumentTypes[i]);
-      paramTypes[i] = paramType == null ? classFor(argumentTypes[i].getClassName()) : paramType;
+      paramTypes[i] = classFor(argumentTypes[i]);
     }
     return Members.methodFor(type, mn.name, paramTypes);
   }
 
+  /**
+   * Returns a origin class for the ASM {@code type}, else {@code null}.
+   */
+  public Class<?> classFor(Type type) {
```

Comment:
I moved this method to another place because it not only resolve for ```primitive``` type.
WDYT.